cmd/fleetctl/query.go

```diff
@@ -123,6 +123,47 @@ func queryCommand() *cli.Command {
 				return errors.New("Query must be specified with --query or --query-name")
 			}
 
+			labels := strings.Split(flLabels, ",")
+			if len(flLabels) > 0 {
+				labelSpecs, err := client.GetLabels()
+				if err != nil {
+					return fmt.Errorf("could not list labels: %w", err)
+				}
+
+				labelsFound := make(map[string]bool)
+				// create a map of labels to check
+				for _, label := range labels {
```

Comment:
You don't need this as accessing a map value for a key that hasn't been set yet will return its zero value (false in this case) IIRC. Double-check me on this.

Reply:
You probably want a map of `struct{}` here to get the behavior you want, as that will allow you to differentiate between unset and set.